src/commands/github/createOctokitClient.ts

```diff
@@ -4,15 +4,34 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { Octokit } from "@octokit/rest";
-import { appendExtensionUserAgent } from "vscode-azureextensionui";
-import { getGitHubAccessToken } from "../../utils/gitHubUtils";
+import { appendExtensionUserAgent, parseError } from "vscode-azureextensionui";
+import { getGitHubAccessToken, logoutOfGitHubSession } from "../../utils/gitHubUtils";
 
 // token should only be passed in during a wizard; otherwise retrieve a new token for the request
 export async function createOctokitClient(token?: string): Promise<Octokit> {
+    const validateCredentials: boolean = !token;
     token = token || await getGitHubAccessToken();
-    return new Octokit(
+    const client: Octokit = new Octokit(
         {
             userAgent: appendExtensionUserAgent(),
             auth: token
         });
+
+    if (validateCredentials) {
+        try {
+            await client.users.getAuthenticated();
+        } catch (error) {
+            if (parseError(error).message.includes('Bad credentials')) {
+                await logoutOfGitHubSession();
```

Comment:
In the issue you said:

> Will direct users to sign out of GitHub command if we encounter this error.

I like that better than logging out _for_ them. It just seems a bit too magic-y, plus the account is managed at the VS Code-level, not just for our extension.

Reply:
Hmm, how should I direct them?  It feels weird explaining it in a notification that the user profiles is in the bottom right corner of their screen.

Reply:
I would like to have a button to sign them out, but I can't find the commandId to execute to initiate the sign out process.  I've looked through their repo and see the actual function, but I don't see anywhere it gets registered and used.

Reply:
Maybe VS Code just doesn't want an extension like us signing users out of something affecting VS Code as a whole. If that's the case, I say we just won't-fix the bug and move on. Like... I think it's fair to expect a "Bad Credentials" error if you revoke access to a key. All you gotta do is sign out and sign back in

Reply:
According to Rachel, other extensions have requested to bring back logout so I'll close this PR.  As for the bug, I do feel like it's a very rare error and the fix is pretty easy, the only issue is it's hard to figure out how to sign out/in.  I think restarting VS Code resolves this problem now though too, so I'm leaning towards just closing it as wont-fix.